cpp/client/ct.cc

```diff
@@ -753,7 +755,7 @@ static void DiagnoseCertChain() {
       continue;
     }
     LOG(INFO) << "SCT number " << i + 1 << ":\n" << sct.DebugString();
-    if (verifier != NULL) {
+    if (verifier.get()) {
```

Comment:
I don't know why, I had a vague feeling you might have had some Control Data in your past life... :smiley:

http://c-faq.com/null/machexamp.html

If you're looking to go against this `if (p)` style, you've got a solid uphill battle ahead of you... For example, because `bool` is an integer (and so, will auto-convert to `int` and friends), objects that want to be "testable" aren't wise to implement an `operator bool`, but instead have a conversion operator to some non-void pointer (usually a private inner class that's never defined, or a method pointer, since those won't bloody convert even to things they ought to!).

I guess those objects would then be comparable with `nullptr`, say, but you're going to have a big pile of existing code that does this (and is perfectly correct to do it).